src/ui/ui_exports.js

```diff
@@ -37,22 +43,43 @@ class UiExports {
     this.consumers.push(consumer);
   }
 
+  // when module ids are defined in a uiExports spec they can
+  // be defined as relative to the public directory
+  resolveModulePath(plugin, moduleId) {
+    const isRelative = moduleId.startsWith('./') || moduleId.startsWith('.\\');
+    if (isRelative) return join(`plugins/${plugin.id}`, moduleId);
+    return moduleId;
+  }
+
+  extendAliases(plugin, alias, idOrIds) {
+    const existingModuleIds = this.aliases[alias] || [];
+    const resolve = id => this.resolveModulePath(plugin, id);
+    const resolvedModuleIds = arr(idOrIds).map(resolve);
+
+    this.aliases[alias] = _.union(existingModuleIds, resolvedModuleIds);
+  }
+
   exportConsumer(type) {
-    for (let consumer of this.consumers) {
+    for (const consumer of this.consumers) {
       if (!consumer.exportConsumer) continue;
-      let fn = consumer.exportConsumer(type);
+      const fn = consumer.exportConsumer(type);
       if (fn) return fn;
     }
 
     switch (type) {
-      case 'app':
       case 'apps':
-        return (plugin, specs) => {
-          for (let spec of [].concat(specs || [])) {
-            let app = this.apps.new(_.defaults({}, spec, {
+        return (plugin, spec) => {
+          for (const rawSpec of arr(spec)) {
```

Comment:
Out of curiosity, why would anything call this function if it wasn't passing in an array `spec`?  The function doesn't seem to do anything unless `spec` is an array.